File: app/src/main/kotlin/metamark/parser/ParenExpressionTerm.kt
package metamark.parser

import metamark.parser.antlr.MetamarkParser

data class ParenExpressionTerm(
    val expression: ExpressionTerm,
): ExpressionTerm() {
    companion object {
        fun build(ctx: MetamarkParser.ParenExpressionContext): ParenExpressionTerm {
            return ParenExpressionTerm(
                expression = ExpressionTerm.build(ctx.wrappedExpression)
            )
        }
    }
}

File: src/main/ParenExpressionTerm.ts
import {ExpressionTerm} from './ExpressionTerm';
import {ParenExpressionContext} from "../generated_parser/MetamarkParser";

export class ParenExpressionTerm extends ExpressionTerm {
    static build(ctx: ParenExpressionContext): ParenExpressionTerm {
        return new ParenExpressionTerm(ExpressionTerm.build(ctx._innerExpression));
    }

    readonly innerExpression: ExpressionTerm;

    constructor(innerExpression: ExpressionTerm) {
        super();
        this.innerExpression = innerExpression;
    }
}
